android/src/main/java/com/sbugert/rnadmob/RNAdMobRewardedVideoAdModule.java

```diff
@@ -0,0 +1,166 @@
+    Callback showAdCallback;
+
+    @Override
+    public String getName() {
+        return "RNAdMobRewarded";
+    }
+
+    public RNAdMobRewardedVideoAdModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public void onRewarded(RewardItem rewardItem) {
+        WritableMap reward = Arguments.createMap();
+
+        reward.putInt("amount", rewardItem.getAmount());
+        reward.putString("type", rewardItem.getType());
+
+        sendEvent("didRewardUserWithReward", reward);
+    }
+
+    @Override
+    public void onRewardedVideoAdLoaded() {
+        sendEvent("rewardedVideoDidLoad", null);
+        requestAdCallback.invoke();
+    }
+
+    @Override
+    public void onRewardedVideoAdOpened() {
+        sendEvent("rewardedVideoDidOpen", null);
+    }
+
+    @Override
+    public void onRewardedVideoStarted() {
+        sendEvent("rewardedVideoDidStart", null);
+    }
+
+    @Override
+    public void onRewardedVideoAdClosed() {
+        sendEvent("rewardedVideoDidClose", null);
+    }
+
+    @Override
+    public void onRewardedVideoAdLeftApplication() {
+        sendEvent("rewardedVideoWillLeaveApplication", null);
+    }
+
+    @Override
+    public void onRewardedVideoAdFailedToLoad(int errorCode) {
+        WritableMap event = Arguments.createMap();
+        String errorString = null;
+
+        switch (errorCode) {
+            case AdRequest.ERROR_CODE_INTERNAL_ERROR:
+            errorString = "ERROR_CODE_INTERNAL_ERROR";
+            break;
+            case AdRequest.ERROR_CODE_INVALID_REQUEST:
+            errorString = "ERROR_CODE_INVALID_REQUEST";
+            break;
+            case AdRequest.ERROR_CODE_NETWORK_ERROR:
+            errorString = "ERROR_CODE_NETWORK_ERROR";
+            break;
+            case AdRequest.ERROR_CODE_NO_FILL:
+            errorString = "ERROR_CODE_NO_FILL";
+            break;
+        }
+
+        event.putString("error", errorString);
+        sendEvent("rewardedVideoDidFailToLoad", event);
+        requestAdCallback.invoke(errorString);
+    }
+
+    private void sendEvent(String eventName, @Nullable WritableMap params) {
+        getReactApplicationContext().getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, params);
+    }
+
+    @ReactMethod
+    public void setAdUnitID(String adUnitID) {
+        this.adUnitID = adUnitID;
+    }
+
+    @ReactMethod
+    public void setTestDeviceID(String testDeviceID) {
+        this.testDeviceID = testDeviceID;
+    }
+
+    @ReactMethod
+    public void requestAd(final Callback callback) {
+        new Handler(Looper.getMainLooper()).post(new Runnable() {
+            @Override
+            public void run () {
+                RNAdMobRewardedVideoAdModule.this.mRewardedVideoAd = MobileAds.getRewardedVideoAdInstance(getCurrentActivity());
+
+                RNAdMobRewardedVideoAdModule.this.mRewardedVideoAd.setRewardedVideoAdListener(RNAdMobRewardedVideoAdModule.this);
```

Comment:
This can go into the constructor in line 34 I think because RewardedVideoAd is a singleton. In the interstitals module I do this because we need to instantiate a new interstitial with each request.